debezium-connector-mysql/src/main/java/io/debezium/connector/mysql/BinlogReader.java

```diff
@@ -74,9 +76,38 @@ public BinlogReader(MySqlTaskContext context) {
         if (logger.isDebugEnabled()) client.registerEventListener(this::logEvent);
 
         // Set up the event deserializer with additional type(s) ...
-        EventDeserializer eventDeserializer = new EventDeserializer();
+        final Map<Long, TableMapEventData> tableMapEventByTableId = new HashMap<Long, TableMapEventData>();
```

Comment:
as per the comment below, this seems like a very laborious and complicated way in which we have to plug in deserialization behavior into the binlog reader